src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_POSITION = 0;
+    private static final int NAME_POSITION = 1;
+    private static final int HOURS_WORKED_POSITION = 2;
+    private static final int SALARY_PER_HOUR_POSITION = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder("Report for period ");
+        stringBuilder.append(dateFrom).append(" - ").append(dateTo);
+        for (String name: names) {
+            int salary = 0;
+            for (String record: data) {
+                if (name.equals(getName(record)) && validateDate(record, dateFrom, dateTo)) {
+                    salary += getDailySalary(record);
```

Comment:
```suggestion
        for (String name: names) {
            int salary = 0;
            for (String record: data) {
                String[] splitedRecord = record.split(" ");
                if (name.equals(splitedRecord[NAME_INDEX]) && validateDate(splitedRecord[DATE_INDEX], dateFrom, dateTo)) {
                    int numberOfHoursWorked =  Integer.parseInt(splitedRecord[HOURS_WORKED_POSITION]);
                    int salaryPerHour = Integer.parseInt(splitedRecord[SALARY_PER_HOUR_POSITION]);
                    salary = numberOfHoursWorked * salaryPerHour;
```
you are splitting the same string 4 times)) lets optimize a little bit